opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -892,17 +969,52 @@ namespace Opm {
 
 
 
-
     template<typename TypeTag>
     bool
     BlackoilWellModel<TypeTag>::
-    assembleImpl(const int iterationIdx,
-                 const double dt,
-                 const std::size_t recursion_level,
-                 DeferredLogger& local_deferredLogger)
+    updateWellControlsAndNetwork(DeferredLogger& local_deferredLogger, const bool balance_network)
     {
+        // not necessarily that we always need to update once of the network solutions
+        // TODO: we do not consider what happens if we could not get network converged
+        // TODO: and we do not consider whether well_group_control_changed affect the network solution
+        // here, there might be something missing in the logic or there should be a better logic
+        // to handle the coupling between the network and well/group solution
+        bool do_network_update = true;
+        bool well_group_control_changed = false;
+        size_t network_iteration = 0;
+        const auto &comm = ebosSimulator_.vanguard().grid().comm();
+        while (do_network_update) {
+            if (network_iteration == 101) {
```

Comment:
The relaxed and hard limits of 100, 200 should be put in named constants, to be used here and below.